plugins/auth-backend/src/lib/oauth/OAuthAdapter.ts

```diff
@@ -78,9 +77,7 @@ export class OAuthAdapter implements AuthProviderRouteHandlers {
     return new OAuthAdapter(handlers, {
       ...options,
       appOrigin,
-      cookieDomain: cookieConfig.domain,
-      cookiePath: cookieConfig.path,
-      secure: cookieConfig.secure,
+      cookieConfig,
```

Comment:
To be honest I think we need to explore a setup where `SameSite=none` is only set in situations where it is absolutely needed, always defaulting to `SameSite=lax` in all other situations. The way I'm thinking we could achieve that is to add `appOrigin` to the cookie configurer input, and then call the configurer in `frameHandler` where we have the actual origin of the app that we're handling authentication for. We won't have that information in any of the other methods, so there we'd just re-use whatever current config the cookie has. We can do this only for the refresh token cookie if that simplifies things.

The cookie config logic would be that `none` if used only if the origin of the backend baseUrl doesn't match the app origin. In the case where a single auth backend handles multiple environments, only one of which is on a matching origin, then a user that heads to the matching origin first would have to log in again when accessing another environment.

I'm feeling that even that above would be a stop-gap for a more elaborate refresh solution that enforces the app origin in a similar way to the OAuth callback handler does.

Reply:
Hey @Rugvip and thanks for taking a look at this! Agree that being more careful about when setting `SameSite=none` makes a lot of sense. 

Let me doublecheck that I understand your suggestion correctly; We can call the cookieConfigurer *again* in the frameHandler passing the `appOrigin` derived from `OAuthState`. In the cookieConfigurer we also determine the value of `SameSite` by comparing the domain from `callbackUrl` (not `baseUrl` right?) with `appOrigin`?
The idea being that we only set `SameSite=None` if the origin of the app is hosted on a different domain than the auth-backend and in all other cases setting `SameSite=Lax`. Does that sum it up?

You did hint about it, but want to make sure that it would be ok with more changes being made here, potentially breaking, than my first attempt? 😅

Reply:
Yep that sounds right! Good call on using `callbackUrl` instead, that probably makes more sense